mr/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergConfigTextParser.java

```diff
@@ -0,0 +1,234 @@
+  }
+
+  @Test
+  public void testParsing1ArgTransformsFailure() {
+    AssertHelpers.assertThrows("should fail when transform not exist",
+        IllegalArgumentException.class,
+        "unknown 1-arg transform",
+        () -> HiveIcebergConfigTextParser.fromPartitioningText(SCHEMA, "unknown(col)"));
+
+    AssertHelpers.assertThrows("should fail when column does not exist",
+        IllegalArgumentException.class,
+        "Cannot find source column: col",
+        () -> HiveIcebergConfigTextParser.fromPartitioningText(SCHEMA, "year(col)"));
+
+    AssertHelpers.assertThrows("should treat additional arguments as plain text column and fail",
+        IllegalArgumentException.class,
+        "unknown 2-arg transform",
+        () -> HiveIcebergConfigTextParser.fromPartitioningText(SCHEMA, "year(8,col)"));
+  }
+
+  @Test
+  public void textMixedTransforms() {
+    PartitionSpec expected = PartitionSpec.builderFor(SCHEMA)
+        .day("created")
+        .bucket("id", 16)
+        .identity("employee_info.employer")
+        .build();
+    Assert.assertEquals(expected, HiveIcebergConfigTextParser.fromPartitioningText(
+        SCHEMA, "day(created)|bucket(16,id)|employee_info.employer"));
+
+    expected = PartitionSpec.builderFor(SCHEMA)
+        .day("created")
+        .identity("employee_info.employer")
+        .bucket("id", 16)
+        .build();
+    Assert.assertEquals(expected, HiveIcebergConfigTextParser.fromPartitioningText(
+        SCHEMA, "day(created)|employee_info.employer|bucket(16,id)"));
+
+    expected = PartitionSpec.builderFor(SCHEMA)
+        .bucket("id", 16)
+        .identity("employee_info.employer")
+        .day("created")
+        .build();
+    Assert.assertEquals(expected, HiveIcebergConfigTextParser.fromPartitioningText(
+        SCHEMA, "bucket(16,id)|employee_info.employer|day(created)"));
+
+    expected = PartitionSpec.builderFor(SCHEMA)
+        .bucket("id", 16)
+        .day("created")
+        .identity("employee_info.employer")
+        .build();
+    Assert.assertEquals(expected, HiveIcebergConfigTextParser.fromPartitioningText(
+        SCHEMA, "bucket(16,id)|day(created)|employee_info.employer"));
+
+    expected = PartitionSpec.builderFor(SCHEMA)
+        .identity("employee_info.employer")
+        .day("created")
+        .bucket("id", 16)
+        .build();
+    Assert.assertEquals(expected, HiveIcebergConfigTextParser.fromPartitioningText(
+        SCHEMA, "employee_info.employer|day(created)|bucket(16,id)"));
+
+    expected = PartitionSpec.builderFor(SCHEMA)
+        .identity("employee_info.employer")
+        .bucket("id", 16)
+        .day("created")
+        .build();
+    Assert.assertEquals(expected, HiveIcebergConfigTextParser.fromPartitioningText(
+        SCHEMA, "employee_info.employer|bucket(16,id)|day(created)"));
+  }
+
+  @Test
+  public void testEscapedColumns() {
+    Schema schema = new Schema(
+        optional(1, "i|d", Types.LongType.get()),
```

Comment:
minor: might want to add an additional case to test other special characters like ";" in column name. Looks like currently we don't require escaping them (e.g. `bucket(16,i;d)` works), I think it might be fine but not sure if we want to make things consistent